### ISSUE TITLE
CKEditor with nested Entry showing up in Blitz Hints

### ISSUE THREAD

**Issue by robzor** [putyourlightson/craft-blitz#714]
### Support Request

Hi Ben!

I've got a Blitz Hint that I'd like to get sorted: "Eager-load the relation field."

The setup:

Matrix field: `entry.cPageContent`
-- Entry type: `bEditorialText`
---- Field (CKEditor): `cEditorialText`
------- Nested Entry Type: `bNestedButton`

So, a matrix field with an editorial block, in which the CKEditor field in there lets you add 'Buttons'.
_(I'm not sure I love these nested entry types, but it's cool that it's a feature now!)_

Do you know where I should be putting the eagerly or the with to make this blitz hint go away?

Not sure if this helps but here's the little route diagram from the Hints plugin:

![CleanShot 2024-09-19 at 7  50 50](https://github.com/user-attachments/assets/4bf9d34a-0003-4718-99f0-af874bb33b4b)

### Plugin Version

5.7.1

**Comment by bencroker** [comment]
Please specify the plugin version whenever creating an issue so I can respond accordingly.

**Comment by bencroker** [comment]
Can you try adding `.eagerly()` any time you query for nested entries? So before each use of `.one()`, `.collection()` and `.all()`.

**Comment by robzor** [comment]
Hi Ben!

So I'm using `eagerly()` everywhere else but the only nested entries thing is in the CKEditor field, which in the template is used like this (where `contentBlocks` is the matrix field and `cEditorialText` is the CKEditor field).

```
{% for block in contentBlocks %}

    {% switch block.type.handle %}

      {# Rich Text #}
      {% case "bEditorialText" %}
        {% include "_partials/entry/bEditorialText" with {
          editorialText: block.cEditorialText
        } only %}

    {% endswitch %}

{% endfor %}
```

If you try and change this line to:  `editorialText: block.cEditorialText.eagerly()` you get this error:

> Neither the property "eagerly" nor one of the methods "eagerly()", "geteagerly()"/"iseagerly()"/"haseagerly()" or "__call()" exist and have public access in class "craft\ckeditor\data\FieldData".

As I'm typing this, the nested entry (a button) has a craft link field (the new one that supports entries) and while I did do some testing with eagerly before, I will have a look into that now again and come back with my findings.

**Comment by robzor** [comment]
Oh for reference, this is the nested entry template (in its current form)

```
{#
    Button (Nested inside CKEditor)
#}

{# Variables #}
{% set buttonLabel = entry.title ??? 'Click here' %}
{% set buttonLink = entry.bNestedButtonLink ??? null %}
{% set buttonLinkNewWindow = entry.bNestedButtonOpenInNewWindow ??? false %}

{% if buttonLink %}

  {% include "_partials/_button" with {
    buttonLabel: buttonLabel,
    buttonLink: buttonLink,
    buttonLinkNewWindow: buttonLinkNewWindow,
    buttonClass: 'button--site-editorial'
  } only %}

{% endif %}
```

**Comment by bencroker** [comment]
I’m not sure whether something can be eager-loaded in your Twig code or whether this is a false positive in Blitz (or whether this is a quirk of having nested entries within nested entries). If you can provide a sample repo that demonstrates this then I can do some investigating, otherwise I’d ignore the hint in the meantime.

**Comment by robzor** [comment]
Ah, so the button link field thing _is_ a red herring, because I can make the Blitz Hint show up if I remove the link field entirely.

So I'm going to say it's definitely to do with the CKEditor using the [clever element rendering thing](https://craftcms.com/docs/5.x/system/elements.html#rendering-elements) and Blitz is throwing a false positive on the `entry` bit within the nested template, imho.

I will ignore for now, and quietly ask for a feature request that individual Hints can be permanently ignored :D

As always, loving your work though! 

Cheers!

**Comment by bencroker** [comment]
Added the ability to ignore hints for the next release in https://github.com/putyourlightson/craft-blitz/commit/adc84af00e0dad95604ff2e82e18d23176065e38.

**Comment by robzor** [comment]
Wow! I wasn't expecting that! Thanks man :)

**Comment by bencroker** [comment]
Released in [5.9.0](https://github.com/putyourlightson/craft-blitz/releases/tag/5.9.0).

**Comment by MattWilcox** [comment]
Oh nice! I'd been having this exact issue and after a few hours decided that it must be a Blitz bug of some sort... I look the next week and it's fixed :D Thanks all.

```
	{% for qAndA in entry.questionsAndAnswers.eagerly().all() %}
		{{ qAndA.render() }}{# Blitz claims this needs eager loading ... but it is eager loaded, see the line above this one #}
	{% endfor %}
```
